src/components/sidebar/RouteShow.tsx

```diff
@@ -9,25 +9,31 @@ import LineHelper from '../../util/lineHelper';
 import TransitTypeColorHelper from '../../util/transitTypeColorHelper';
 import ColorScale from '../../util/colorScale';
 import * as s from './routeShow.scss';
-import { History, Location } from 'history';
-import LinkBuilder from '../../factories/linkBuilder';
+import routeBuilder from '../../routing/routeBuilder';
+import navigator from '../../routing/navigator';
+import QueryParams from '../../routing/queryParams';
 
 interface IRouteShowProps {
     routeStore?: RouteStore;
     route: IRoute;
     visibleRoutePathsIndex: number;
-    history: History;
-    location: Location<any>;
 }
 
 @inject('routeStore')
 @observer
 class RouteShow extends React.Component<IRouteShowProps> {
+    constructor(props: IRouteShowProps) {
+        super(props);
+        this.closeRoute = this.closeRoute.bind(this);
+    }
 
-    private onClose = () => {
+    private closeRoute() {
         this.props.routeStore!.removeFromRoutes(this.props.route.routeId);
-        this.props.history.push(LinkBuilder
-            .createLinkWithoutRoute(this.props.location, this.props.route.routeId));
+        const closeRouteLink = routeBuilder
+            .current()
+            .remove(QueryParams.routes, this.props.route.routeId)
```

Comment:
Why call .current()?
Use instead:
routeBuilder.remove().toLink();

Compare with:
routeBuilder.to(subSites.home).toLink();